### DISCUSSION TITLE
Solar production total - manual calibration

### DISCUSSION THREAD

**Discussion by Byte-M3** [robbinjanssen/home-assistant-omnik-inverter#236] [Q&A]
Thanks for creating this! I just installed it, it's working on an omniksol-5k-tl2 via JS.

I already have my solar panels for a couple of year, so I would like to adjust the solar production total value. Is that possible?

**Comment by robbinjanssen** [comment]
The total product value should be outputted by the inverter itself? Can you share your output? :-)

**Reply by Byte-M3** [reply]
Hi Robbin,

Not sure if this is the output you mean, but let me know if you need something else.
![omnik](https://github.com/robbinjanssen/home-assistant-omnik-inverter/assets/2970732/67337722-8ed6-4443-917c-739dbfac0c09)

When i posted the question, the numbers were also identical to each other, and I assumed that the summation needed to occur within home assistant. But now I see that there is no history kept in HA, so it seems that the Solar production total sensor is not getting the correct input from the inverter.

**Reply by robbinjanssen** [reply]
Seems like your inverter is giving some other data then the integration expects. Which method do you use to fetch the data? JS, HTML, JSON?

**Reply by Byte-M3** [reply]
I used JS, figured it was working because I got data.

**Reply by robbinjanssen** [reply]
Can you provide me with the output of `http://<your inverter ip>/js/status.js`?

**Reply by Byte-M3** [reply]
```
var version= "H4.01.51MW.2.01W1.0.64(2018-01-251-D)";
var m2mMid = "626726032";
var wlanMac= "BC:54:F9:F7:36:80";
var m2mRssi= "44%";
var wanIp= "192.168.0.112";
var nmac= "BC54F9F73683";
var fephy= "off";
var yz_device_type="258";
var yz_device_num="1";
var myDeviceArray=new Array();
myDeviceArray[0]="NLBN5020189T1093,V5.27Build261,V5.43Build182,omnik5000tl2,5000,1154,1304,130,,2,";
;
... removed;
```

**Reply by robbinjanssen** [reply]
So the part we parse is actually just this little bit;
`myDeviceArray[0]="NLBN5020189T1093,V5.27Build261,V5.43Build182,omnik5000tl2,5000,1154,1304,130,,2,";`

It seems that it doesn't include the grand total of your inverter?
Does your inverter display or web interface show the total somewhere?

**Reply by Byte-M3** [reply]
Hi Robbin, thanks for the support :)

I checked the web interface, and it gives this output:
![image](https://github.com/robbinjanssen/home-assistant-omnik-inverter/assets/2970732/ec0e3753-8010-4780-a767-08fc39150496)
So it actually looks like your integration works correctly, but my omnik provides the wrong data, because I get exactly the same numbers from the omnik sensors in HA:
![image](https://github.com/robbinjanssen/home-assistant-omnik-inverter/assets/2970732/f6251c26-4faa-4a8b-bdf9-59f58d97c3a8)
Too bad, but the daily and current output are more important anyway. I can get the total from the web monitoring interface provided by the installation company (although also not directly, but I can get totals for each year).

**Comment by dadalouis** [comment]
![Scherm­afbeelding 2024-06-13 om 22 00 56](https://github.com/robbinjanssen/home-assistant-omnik-inverter/assets/89429994/cd2f956c-85a6-4489-ae56-17866e64b3e3)
im was wondering the same. 
i just got a used one and the value is already 5400 kw and i would like to reset the value to 0. just in ha for my own record.
